### ISSUE TITLE
[Bug]: App Name is not available when using Portal Notifications

### ISSUE THREAD

**Issue by Crylia** [flatpak/xdg-desktop-portal#916]

**Comment by GeorgesStavracas** [comment]
Please be more specific. Which specific apps have you used that trigger this behavior, and what did you expect to see?

**Comment by Crylia** [comment]
One example would be spotify, I would expect a notification to also contain the applications name, but it doesn't.

https://specifications.freedesktop.org/notification-spec/latest/ar01s09.html The freedesktop spec defines a "app_name" which is not set when using any application from within flatpak

**Comment by swick** [comment]
Is the app flatpak, snap or host? Which portal backend do you use?

**Comment by Crylia** [comment]
It would be a flatpak, a good example would be Spotify (app/com.spotify.Client/x86_64/stable)

**Comment by swick** [comment]
This is very likely an issue with a specific backend + compositor. Which ones do you use? I suspect xdg-desktop-portal-gtk and not gnoms-shell?

https://github.com/flatpak/xdg-desktop-portal-gtk/blob/366d3349041c2e06658c7c5a8b5bfd5f345d8157/src/fdonotification.c#L348

**Comment by Crylia** [comment]
How do I check/change the backend? I just use awesomewm with picom

**Comment by swick** [comment]
`ps ax| grep xdg-desktop-portal` might be sufficient, otherwise check which component supports the notification interface in `/usr/share/xdg-desktop-portal/portals/` or use `dbus-monitor` to figure out what gets called.

**Comment by Crylia** [comment]
```
3239 ?        SNsl   0:00 /usr/libexec/xdg-desktop-portal
   3256 ?        SNsl   0:00 /usr/libexec/xdg-desktop-portal-gnome
   3265 ?        SNsl   0:00 /usr/libexec/xdg-desktop-portal-gtk
   3270 ?        SNsl   0:00 /usr/lib/x86_64-linux-gnu/libexec/xdg-desktop-portal-kde
  46431 pts/0    SN+    0:00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox xdg-desktop-portal
  ```
  is the output from the first command, so ig I got three running?

**Comment by swick** [comment]
Yes, each can provide a number of portals. the gnome backend doesn't implement notifications, gtk does and I don't know about KDE (it probably does). You can check in /usr/share/xdg-desktop-portal/portals/kde.portal if it implements the notification Interface.

If both gtk and kde do then you can use dbus-monitor "interface='org.freedesktop.impl.portal.Notification'" to figure out which one gets called.

**Comment by georg-mayer** [comment]
Hi, I have the exact same problem - it is with a flatpak spotify installation on kbuntu (KDE). The error is annoying, as an error sound is played after every song. From the answers so far I don't know how to resolve the problem. Did I miss something? Thanks for any help on this!

**Comment by ray-kast** [comment]
Seconding this, I've been using `xdg-desktop-portal-gtk` with i3 for awhile and after a recent update to `xdg-desktop-portal@1.18.1` and `xdg-desktop-portal-gtk@1.15.1` notifications no longer have an attached app name.  I've confirmed from using the above `dbus-monitor` command that `xdg-desktop-portal-gtk` is receiving the notifications.

**Comment by rpetti** [comment]
Noticed this with both the firefox and vencord flatpaks. Looks like app name isn't even a field in the portal notification spec, and there's no handling of it in the code, so I assume that's why it's coming across empty.

**Comment by A6GibKm** [comment]
The error comes from libnotify: https://gitlab.gnome.org/GNOME/libnotify/-/blob/master/libnotify/notification.c#L479. This is because the app tries to tell libnotify the app name but this feature is not available if using the notification portal. Note that the app name is something that the backend and xdp should know, not something the app should declare.
